Shellcheck 0.4.6 new errors cause make check to fail

```
==> Starting check()...
In bin/aurchain line 55:
        if [[ ${target[@]} ]]; then
              ^-- SC2199: Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

In bin/aursift line 34:
    siftargs+=(--satisfies="$p")
               ^-- SC2191: The = here is literal. To assign by index, use ( [index]=value ) with no spaces. To keep as literal, quote it.

In bin/aursift line 38:
if [[ ${pkgset[@]} ]]; then
      ^-- SC2199: Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

In bin/aursift line 40:
        siftargs+=(--repo="$p")
                   ^-- SC2191: The = here is literal. To assign by index, use ( [index]=value ) with no spaces. To keep as literal, quote it.

In bin/aursift line 50:
if [[ ${query[@]} ]]; then
      ^-- SC2199: Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

In bin/aursync line 129:
    if [[ ${avail[@]} ]]; then
          ^-- SC2199: Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

make: *** [Makefile:9: check] Error 1
==> ERROR: A failure occurred in check().
    Aborting...
```
If I understand the intent of the code correctly, the SC2199 errors can all be fixed by using * instead of @, and SC2199 warning can be avoided by writing as `"--param=$var"` in those places

--- a/bin/aurchain
+++ b/bin/aurchain
@@ -52,7 +52,7 @@ chain() {
         # Avoid querying duplicates (#4)
         readarray -t target < <(fld 2 tsv/$sub | grep -Fxvf seen)
 
-        if [[ ${target[@]} ]]; then
+        if [[ ${target[*]} ]]; then
             aursearch -Fr -- "${target[@]}" > json/$a || exit
         else
             break

--- a/bin/aursift
+++ b/bin/aursift
@@ -31,13 +31,13 @@ fi
 readarray -t all
 
 for p in "${all[@]}"; do
-    siftargs+=(--satisfies="$p")
+    siftargs+=("--satisfies=$p")
     pkgset[$p]=1
 done
 
-if [[ ${pkgset[@]} ]]; then
+if [[ ${pkgset[*]} ]]; then
     for p in "${arch_repo[@]}"; do
-        siftargs+=(--repo="$p")
+        siftargs+=("--repo=$p")
     done
 else
     error "$argv0: no data available"
@@ -47,7 +47,7 @@ fi
 # https://github.com/andrewgregory/pacutils/issues/3
 readarray -t query < <(pacsift --sync --exact "${siftargs[@]}" <&-)
 
-if [[ ${query[@]} ]]; then
+if [[ ${query[*]} ]]; then
     while read -r p v; do
         p=${p#*/}
 

--- a/bin/aursync
+++ b/bin/aursync
@@ -132,7 +132,7 @@ else
         fi
     done < <(pacconf --repo-list)
 
-    if [[ ${avail[@]} ]]; then
+    if [[ ${avail[*]} ]]; then
         if [[ ${#avail[@]} -gt 1 ]]; then
             PS3='Select a repository: '
             select repo in "${!avail[@]}"; do

